debezium-core/src/test/java/io/debezium/transforms/TimezoneConverterTest.java

```diff
@@ -0,0 +1,848 @@
+        SourceRecord customersRecord = new SourceRecord(
+                new HashMap<>(),
+                new HashMap<>(),
+                "db.server1.customers",
+                customersEnvelope.schema(),
+                customersPayload);
+
+        final SourceRecord transformedCustomersRecord = converter.apply(customersRecord);
+        final Struct transformedCustomersValue = (Struct) transformedCustomersRecord.value();
+        final Struct transformedCustomersAfter = transformedCustomersValue.getStruct(Envelope.FieldName.AFTER);
+        assertThat(transformedCustomersAfter.get("name")).isEqualTo("Amy Rose");
+        assertThat(transformedCustomersAfter.get("order_time")).isEqualTo("10:19:25+00:00");
+
+        Schema inventoryRecordSchema = SchemaBuilder.struct()
+                .field("id", Schema.INT8_SCHEMA)
+                .field("name", Schema.STRING_SCHEMA)
+                .field("shipping_time", ZonedTime.builder().optional().build())
+                .field("order_time", ZonedTime.builder().optional().build())
+                .build();
+
+        final Struct inventoryBefore = new Struct(inventoryRecordSchema);
+        inventoryBefore.put("id", (byte) 1);
+        inventoryBefore.put("name", "Amy Rose");
+        inventoryBefore.put("shipping_time", "19:19:25+00:00");
+        inventoryBefore.put("order_time", "10:19:25+00:00");
+
+        final Struct inventorySource = new Struct(sourceSchema);
+        inventorySource.put("table", "inventory");
+        inventorySource.put("lsn", 1);
+        inventorySource.put("ts_ms", 123456789);
+
+        final Envelope inventoryEnvelope = Envelope.defineSchema()
+                .withName("dummy.Envelope")
+                .withRecord(inventoryRecordSchema)
+                .withSource(sourceSchema)
+                .build();
+
+        final Struct inventoryPayload = inventoryEnvelope.create(inventoryBefore, inventorySource, Instant.now());
+        SourceRecord inventoryRecord = new SourceRecord(
+                new HashMap<>(),
+                new HashMap<>(),
+                "db.server1.inventory",
+                inventoryEnvelope.schema(),
+                inventoryPayload);
+
+        final SourceRecord transformedInventoryRecord = converter.apply(inventoryRecord);
+        final Struct transformedInventoryValue = (Struct) transformedInventoryRecord.value();
+        final Struct transformedInventoryAfter = transformedInventoryValue.getStruct(Envelope.FieldName.AFTER);
+        assertThat(transformedInventoryAfter.get("name")).isEqualTo("Amy Rose");
+        assertThat(transformedInventoryAfter.get("shipping_time")).isEqualTo("22:19:25+03:00");
+        assertThat(transformedInventoryAfter.get("order_time")).isEqualTo("10:19:25+00:00");
+    }
+
+    @Test
+    public void testDayLightSavings() {
+        String withDayLightSavingsInstant = "2014-12-22T10:15:30Z"; // DST is in effect
```

Comment:
These two values are strange - december is definitely not DST and June is DST.

Reply:
Sorry for the confusion, values got interchanged, otherwise works fine! :)